src/spyglass/spikesorting/v1/sorting.py

```diff
@@ -242,18 +243,40 @@ def make(self, key: dict):
             # Specify tempdir (expected by some sorters like mountainsort4)
             sorter_temp_dir = tempfile.TemporaryDirectory(dir=temp_dir)
             sorter_params["tempdir"] = sorter_temp_dir.name
+            os.chmod(temp_dir_path, 0o777)
+
+            if sorter == "mountainsort5":
+                _ = sorter_params.pop("tempdir", None)
+
             # if whitening is specified in sorter params, apply whitening separately
             # prior to sorting and turn off "sorter whitening"
-            if sorter_params["whiten"]:
+            if sorter_params.get("whiten", False):
                 recording = sip.whiten(recording, dtype=np.float64)
                 sorter_params["whiten"] = False
-            sorting = sis.run_sorter(
-                sorter,
-                recording,
-                output_folder=sorter_temp_dir.name,
-                remove_existing_folder=True,
-                **sorter_params,
-            )
+
+            common_sorter_items = {
+                "sorter_name": sorter,
+                "recording": recording,
+                "output_folder": sorter_temp_dir.name,
+                "remove_existing_folder": True,
+            }
+            if sorter.lower() in ["kilosort2_5", "kilosort3", "ironclust"]:
```

Comment:
@donghoon-shin can you verify that `kilosort2_5` the name of the sorter instead of e.g. `kilosort2.5` etc?

Reply:
verified!